### ISSUE TITLE
Allow podman to run Helm charts

### ISSUE THREAD

**Issue by chargio** [containers/podman#15098]
<!--
---------------------------------------------------
BUG REPORT INFORMATION
---------------------------------------------------
**Is this a BUG REPORT or FEATURE REQUEST? (leave only one on its own line)**

/kind feature

**Description**

Podman can run different yaml files from Kubernetes easily. I would like to be able to run a helm chart, so several of those are run concurrently and podman can be used for development.
I would love to have a way of running a postgresql container with an external storage as a helm chart, and manage it through podman, so I don't need to use a virtualization layer.
This will allow me to start and stop a full environment with a single comand instead of having to manuall start different components.

**Describe the results you expected:**
I would expect to be able to specify a helm chart and be able to execute it in a way similar to what is supported by kube play.

**Package info (e.g. output of `rpm -q podman` or `apt list podman`):**
Running on a Mac M1 with macOS Monterey

**Have you tested with the latest version of Podman and have you checked the Podman Troubleshooting Guide? (https://github.com/containers/podman/blob/main/troubleshooting.md)**

Yes

**Comment by flouthoc** [comment]
Hi @chargio , Thanks for creating the issue.
 
I used to use `helm template --dry-run --debug . --generate-name --values values.yaml` to generate kube yamls. I am sure generated yamls can be used with `podman` as well except for some objects since not all kubernetes objects are implemented in `podman` afaik.

PS: I used to use this command few moths ago and **not tested with podman** so not sure if helm supports this anymore or not ( src: https://gist.github.com/flouthoc/abbd41435b3b5f84c96de9deaf20edb7 )

But i'd wait for @containers/podman-maintainers 's take on this.

**Comment by chargio** [comment]
Yes, i was trying to set up my development laptop and being able to use a helm chart directly looked like a great way of setting it up, knowing that not all the element will be created, but being as close to production as possible

**Comment by rhatdan** [comment]
If you wanted to open a WIP PR to show us how this would work?
Perhaps

podman kube play --help APPLICATION.HELM

BTW, I no nothing about HELM Charts.

**Comment by vrothberg** [comment]
I think we should first discuss whether Podman should/can support Helm charts.  Just to spare @chargio or others from doing redundant work.

I do not know Helm at all beyond having skimmed the website (https://helm.sh/docs/intro/using_helm/). To me, it feels out of scope.

Is there a way to generate K8s YAML from Helm?

**Comment by rhatdan** [comment]
I know even less.  We need people with K8s experience to discuss. @mrunalp @haircommander @umohnani8 @saschagrunert @kolyshkin Thoughts?

**Comment by haircommander** [comment]
to me, it makes sense to approach this the way @flouthoc did: convert the helm chart to kube yaml and apply those. That's effectively what helm does in an actual kube cluster, so it would leave podman in the same place in the stack.

**Comment by rhatdan** [comment]
So you are saying the user should convert the helm chart, and Podman would just consume kubernetes.YAML, as opposed to Podman supporting helm chargs and then converting them internally to kubernetes.YAML.

**Comment by haircommander** [comment]
yeah!

**Comment by vrothberg** [comment]
> So you are saying the user should convert the helm chart, and Podman would just consume kubernetes.YAML, as opposed to Podman supporting helm chargs and then converting them internally to kubernetes.YAML.

If that works, awesome!

**Comment by flouthoc** [comment]
@vrothberg it should atleast on the older versions it used to work: https://github.com/containers/podman/issues/15098#issuecomment-1197957136

**Comment by carljmosca** [comment]
I really appreciate this discussion...given that (as original issue is references) the [helm template command ](https://helm.sh/docs/helm/helm_template/) generates kubernetes yaml from a helm chart, there should not be too much needed here (unless I am missing something).  I agree that putting Helm-aware functionality inside podman is/should be out-of-scope.  

I think a list of kubernetes-like functionality unavailable in podman/differences in implementation and of that list what should and should not be supported.  For example, secrets, as I understand it, can be created using podman but is a "kube kind" secret intended?  Perhaps this already exists and I have missed it.

Thank you for this [article](https://www.redhat.com/sysadmin/kubernetes-workloads-podman-systemd) by @rhatdan and others as a reminder to try a use case or two (this approach has been off my radar since learning of it but I can think of some excellent use cases).

**Comment by vrothberg** [comment]
I am going to close as I consider the issue to be resolved.  Thanks a lot for the discussion!
